src/main/java/com/example/weatherwaveapi/controller/WeatherController.java

```diff
@@ -17,26 +20,38 @@
 @Slf4j
 public class WeatherController {
     private static final String REQUEST_MESSAGE = "Request contain serviceApiEnumValue: {}";
+    private static final String INVALID_INPUT = "Invalid input for parameter";
+    private static final String PROVIDE_VALID_VALUE = ". Please provide a valid value.";
 
     private final WeatherService weatherService;
 
-    @GetMapping(value = "/city", produces = MediaType.APPLICATION_JSON_VALUE)
-    public ResponseEntity<WeatherResponse> getWeatherInSelectedCity(@RequestParam String city) {
-        WeatherRequest weatherRequest = WeatherRequest.builder()
-                .cities(List.of(city))
-                .build();
-        log.info(REQUEST_MESSAGE, weatherRequest);
-        WeatherResponse weatherResponse = weatherService.getWeather(weatherRequest);
-        return ResponseEntity.ok(weatherResponse);
+    @GetMapping(value = "/{city}", produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<WeatherResponse> getWeatherInSelectedCity(@PathVariable(name = "city") String city,
+                                                                    @RequestParam(name = "serviceApi", defaultValue = "OPEN_WEATHER_MAP", required = false) ServiceApiEnum inputOfServiceApi) {
+        return getWeatherResponse(List.of(city), inputOfServiceApi);
     }
 
     @GetMapping(value = "/cities", produces = MediaType.APPLICATION_JSON_VALUE)
-    public ResponseEntity<WeatherResponse> getWeatherInSelectedCities(@RequestParam List<String> cities) {
+    public ResponseEntity<WeatherResponse> getWeatherInSelectedCities(
+            @RequestParam(value = "cities") List<String> cities,
+            @RequestParam(value = "serviceApi", defaultValue = "OPEN_WEATHER_MAP", required = false) ServiceApiEnum inputOfServiceApi
+    ) {
+        return getWeatherResponse(cities, inputOfServiceApi);
+    }
+
+    @ExceptionHandler(MethodArgumentTypeMismatchException.class)
+    @ResponseStatus(HttpStatus.BAD_REQUEST)
+    public ResponseEntity<String> handleMethodArgumentTypeMismatch(MethodArgumentTypeMismatchException ex) {
+        String errorMessage = INVALID_INPUT + ex.getName() + PROVIDE_VALID_VALUE;
+        return ResponseEntity.status(HttpStatus.BAD_REQUEST).body(errorMessage);
+    }
+
+    private ResponseEntity<WeatherResponse> getWeatherResponse(List<String> cities, ServiceApiEnum serviceApi) {
         WeatherRequest weatherRequest = WeatherRequest.builder()
                 .cities(cities)
+                .service(serviceApi)
                 .build();
-        log.info(REQUEST_MESSAGE, weatherRequest);
-        WeatherResponse response = weatherService.getWeather(weatherRequest);
-        return ResponseEntity.ok(response);
+
+        log.info(REQUEST_MESSAGE, weatherRequest.service());
```

Comment:
Sorry I don't understand, there is no return value in this method

Reply:
I guess it was in this way because of rebasing. But I fixed it